src/pymor/models/transfer_function.py

```diff
@@ -0,0 +1,473 @@
+# This file is part of the pyMOR project (https://www.pymor.org).
+# Copyright 2013-2021 pyMOR developers and contributors. All rights reserved.
+# License: BSD 2-Clause License (https://opensource.org/licenses/BSD-2-Clause)
+
+import numpy as np
+import scipy.linalg as spla
+
+from pymor.algorithms.to_matrix import to_matrix
+from pymor.core.cache import CacheableObject
+from pymor.core.cache import cached
+from pymor.parameters.base import ParametricObject, Mu
+
+
+class TransferFunction(CacheableObject, ParametricObject):
+    r"""Class for systems represented by a transfer function.
+
+    This class describes input-output systems given by a transfer
+    function :math:`H(s, \mu)`.
+
+    Parameters
+    ----------
+    dim_input
+        The number of inputs.
+    dim_output
+        The number of outputs.
+    tf
+        The transfer function defined at least on the open right complex half-plane.
```

Comment:
It is an assumption that `TFIRKAReductor` uses. But, yeah, it would make more sense to move this assumption there. Also, `h2_norm` needs values on the imaginary axis...